dds/DCPS/Serializer.inl

```diff
@@ -841,18 +838,32 @@ Serializer::align_cont_w()
 ACE_INLINE
 bool Serializer::read_delimiter(size_t& size)
 {
-  ACE_CDR::ULong dheader;
-  if (!(*this >> dheader)) {
-    return false;
+  if (encoding().xcdr_version() == Encoding::XCDR_VERSION_2) {
+    ACE_CDR::ULong dheader;
+    if (!(*this >> dheader)) {
+      return false;
+    }
+    size = dheader;
   }
-  size = dheader;
   return true;
 }
 
 ACE_INLINE
 bool Serializer::write_delimiter(size_t size)
 {
-  return *this << static_cast<ACE_CDR::ULong>(size);
+  if (encoding().xcdr_version() == Encoding::XCDR_VERSION_2) {
+    return *this << static_cast<ACE_CDR::ULong>(size - uint32_cdr_size);
```

Comment:
> Looks like read/write delimiter do this differently so they don't "round trip".

Yes, this reflects usage because the delimiter is always included by `serialized_size` when we write a delimiter, while we don't need that information while reading. I can include that information in the documenting comment.